main.py

```diff
@@ -0,0 +1,74 @@
+import sys
+import json
+
+def gettime(timeanddate):
+  return timeanddate[11:16]
+
+def samelocation(previous, current):
+  plat = previous['latitude']
+  plng = previous['longitude']
+  clat = current['latitude']
+  clng = current['longitude']
+  if(plat == clat and plng == clng):
+    return True
+  return False
+
+def mapfunction(data):
+  response = []
+  data = data['data']
+  previousRow = ''
+  curpath = []
+  curtype = ''
+  starttime = ''
+  endtime = ''
+  startvol = 0
+  endvolume = 0
+  volumeadded = 0
+  volumeused = 0
+  if len(data) > 0:
+    for row in data:
+      if previousRow != '':
+        if (samelocation(previousRow, row) and (curtype == 'stop' or curtype == '')):
+          curtype = 'stop'
+        elif (samelocation(previousRow, row) and curtype == 'moving'):
+          curpath.append({'lat': float(row['latitude']), 'lng': float(row['longitude'])})
+          response.append({'type': curtype, 'path': curpath, 'starttime': starttime, 'endtime': endtime, 'voa': startvol, 'vol': endvolume, 'va': volumeadded, 'vu': volumeused})
+          startvol = float(previousRow['volume_left'])
```

Comment:
Codacy found an issue: [Sequence index is not an int, slice, or instance with __index__](https://app.codacy.com/gh/UptimaNGR/uptima-backend/pullRequest?prid=6279797)